### ISSUE TITLE
`@by` not using optimised `by` from DataFrames.jl

### ISSUE THREAD

**Issue by xiaodaigh** [JuliaData/DataFramesMeta.jl#135]
See MEW, DataFramesMeta is 2x slower.

```julia
using DataFrames, DataFramesMeta, RDatasets, BenchmarkTools
a  = dataset("datasets", "iris")

@btime xres = by(a, :Species, :Species => length);
# 21.300 μs (159 allocations: 14.64 KiB)

@btime @by(a, :Species, lc = length(:Species))
#  40.699 μs (306 allocations: 25.50 KiB)
```

**Comment by nalimilan** [comment]
Yes, currently it creates a single anonymous function for all pairs. Instead, it should create one anonymous function for each pair, and pass them using the pair syntax of `by`. That shouldn't be too hard.
See also https://github.com/JuliaData/DataFramesMeta.jl/pull/127#issuecomment-476380700 where I gave some pointers.

**Comment by robsmith11** [comment]
This is realllly needed.  For a medium size table with 12 million rows and a simple operation on two Int64 columns, it's nearly 20x slower than plain DataFrames' `by`:

```
julia> @btime by($df, :id, :time => length);
  595.707 ms (175 allocations: 725.20 MiB)
                                      
julia> @btime @by($df, :id, n=length(:time))
  21.664 s (290410814 allocations: 19.30 GiB)

```

**Comment by bkamins** [comment]
@robsmith11 - thank you for commenting on the issues with DataFramesMeta.jl.
As you probably can notice - this package is currently not very actively developed currently.

Therefore - if you would find time for it and be willing to - feel free to make PRs fixing the issues you think are worth improving. I can review these PRs for you.

**Comment by robsmith11** [comment]
Ok, thanks for the status update.  I'll take a look when I have time to get familiar with the code base.

I have some ideas from kdb+/q's sql syntax that I'd like to try as well, so I might end up creating a fork with which to experiment.

**Comment by bkamins** [comment]
Sure - you can also ping me on the fork to have a look.

**Comment by pdeffebach** [comment]
Fixed by #163. We no longer implement our own `by` behavior.
